designs/karpenter-integration-testing.md

```diff
@@ -0,0 +1,244 @@
+
+##### Expiration
+
+* High Level Description: Validate expiration operation for karpenter. This should validate that nodepool disruption budget are not respected, however PDBs are respected   
+* Source: https://github.com/aws/karpenter-provider-aws/tree/main/test/suites/expiration 
+* Test Cases: 
+    * should expire the node after the expiration is reached
+    * should replace expired node with a single node and schedule all pods 
+
+##### Integration
+
+* High Level Description: Multiple functionality are tested. This is the catch all suite for testing karpenter
+* Source: https://github.com/aws/karpenter-provider-aws/tree/main/test/suites/integration
+* Test Cases: 
+    * DaemonSet
+        * should account for LimitRange Default on daemonSet pods for resources
+        * should account for LimitRange DefaultRequest on daemonSet pods for resources
+    * Hash
+        * should have NodePool hash
+    * Repair Policy
+        * should ignore disruption budgets
+        * should ignore do-not-disrupt annotation on node
+        * should ignore terminationGracePeriod on the nodepool
+    * Utilization
+        * should provision one pod per node
+    * Validation 
+        * NodePool
+            * should error when a restricted label is used in labels (karpenter.sh/nodepool)
+            * should error when a restricted label is used in labels (kubernetes.io/custom-label)
+            * should allow a restricted label exception to be used in labels (node-restriction.kubernetes.io/custom-label)
+            * should allow a restricted label exception to be used in labels ([*].node-restriction.kubernetes.io/custom-label)
+            * should error when a requirement references a restricted label (karpenter.sh/nodepool)
+            * should error when a requirement uses In but has no values
+            * should error when a requirement uses an unknown operator
+            * should error when Gt is used with multiple integer values
+            * should error when Lt is used with multiple integer values
+            * should error when consolidateAfter is negative
+            * should succeed when ConsolidationPolicy=WhenEmptyOrUnderutilized is used with consolidateAfter
+            * should error when minValues for a requirement key is negative
+            * should error when minValues for a requirement key is zero
+            * should error when minValues for a requirement key is more than 50
+            * should error when minValues for a requirement key is greater than the values specified within In operator
+
+##### Nodeclaim
+
+* High Level Description: Validate that action taken by the nodeclaim controllers. This validates standalone claims and the garbage collection mechanisms  
+* Source: https://github.com/aws/karpenter-provider-aws/tree/main/test/suites/nodeclaim
+* Test Cases: 
+    * Standalone NodeClaim 
+        * should create a standard NodeClaim with the c instance family 
```

Comment:
what is the "c instance family", is this an AWS specific reference?

Reply:
Yeah, this is a AWS specific reference. The idea here would that we can launch any generic instance type. I will drop this test since we will get this validation for free with the other tests